### PR TITLE
[CoreData] Add support for xcode 15 beta6.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: src/CoreData/Enums.cs [hunk 1]
```diff
@@ -58,7 +58,9 @@ public enum NSAttributeType : ulong {
 		[MacCatalyst (13, 1)]
 		Uri = 1200,
 		Transformable = 1800,
-		ObjectID = 2000
+		ObjectID = 2000,
+		[Watch (7, 0), TV (14, 0), Mac (11, 0), iOS (14, 0), MacCatalyst (17, 0)]
```
Comment (tj-devel709) [inline src/CoreData/Enums.cs:62]: Why old versions here?

https://developer.apple.com/documentation/coredata/nsattributetype/compositeattributetype?changes=__6

File: src/coredata.cs [hunk 2]
```diff
@@ -2346,6 +2364,14 @@ partial interface NSPersistentStoreCoordinator
...
 		[return: NullAllowed]
 		NSPersistentHistoryToken GetCurrentPersistentHistoryToken ([NullAllowed] NSObject [] stores);
+
+		[Watch (7, 0), TV (14, 0), Mac (11, 0), iOS (14, 0)]
+		[Export ("finishDeferredLightweightMigration:")]
+		bool FinishDeferredLightweightMigration ([NullAllowed] out NSError error);
+
+		[Watch (7, 0), TV (14, 0), Mac (11, 0), iOS (14, 0)]
+		[Export ("finishDeferredLightweightMigrationTask:")]
+		bool FinishDeferredLightweightMigrationTask ([NullAllowed] out NSError error);
```
Comment (tj-devel709) [inline src/coredata.cs:2374]: These are old versions too but looks like they may actually be older? Just checking

File: src/CoreData/Enums.cs (hunk 1 above)
Comment (mandel-macaque) [reply src/CoreData/Enums.cs:62]: due to a yy.

File: src/coredata.cs (hunk 2 above)
Comment (mandel-macaque) [reply src/coredata.cs:2374]: This ones are correct. We probably forgot them in one of the . release.

**Review by haritha-mohan — APPROVED** [review]

**Pull request merged**